src/DotLiquid/Util/StrFTime.cs

```diff
@@ -16,87 +26,199 @@ public static class StrFTime
+        public static string ToStrFTime(this DateTime dateTime, string format)
         {
-            string output = "";
+            return Regex.Replace(input: format, pattern: SPECIFIER_REGEX,
+                evaluator: specifier => SpecifierEvaluator(
+                    specifier: specifier.Groups[0].Value,
+                    flags: specifier.Groups[GROUP_FLAGS].Captures.Cast<Capture>().Select(capture => capture.Value).ToList(),
+                    width: specifier.Groups[GROUP_WIDTH].Captures.Cast<Capture>().Select(capture => (int?)Convert.ToInt32(capture.Value)).FirstOrDefault(),
+                    directive: specifier.Groups[GROUP_DIRECTIVE].Captures.Cast<Capture>().Select(capture => capture.Value).FirstOrDefault(),
+                    source: dateTime
+                    ));
+        }
 
-            int n = 0;
+        /// <summary>
+        /// Formats a date using a ruby date format string
+        /// </summary>
+        /// <param name="dateTimeOffset">date-time object to be formatted</param>
+        /// <param name="format">the required format</param>
+        /// <returns>a string version of date-time matching pattern.</returns>
+        public static string ToStrFTime(this DateTimeOffset dateTimeOffset, string format)
+        {
+            return Regex.Replace(input: format, pattern: SPECIFIER_REGEX,
+                evaluator: specifier => SpecifierEvaluator(
+                    specifier: specifier.Groups[0].Value,
+                    flags: specifier.Groups[GROUP_FLAGS].Captures.Cast<Capture>().Select(capture => capture.Value).ToList(),
+                    width: specifier.Groups[GROUP_WIDTH].Captures.Cast<Capture>().Select(capture => (int?)Convert.ToInt32(capture.Value)).FirstOrDefault(),
+                    directive: specifier.Groups[GROUP_DIRECTIVE].Captures.Cast<Capture>().Select(capture => capture.Value).FirstOrDefault(),
+                    source: dateTimeOffset
+                    ));
+        }
 
-            while (n < pattern.Length)
-            {
-                string s = pattern.Substring(n, 1);
+        /// <summary>
+        /// Formats a date-time for a single specifier within the requested format. For ease of processing
+        /// the specifier is broken down into it's consituent parts of: flags, width, directive.
+        ///
+        /// A specifier consists of a percent (%) character, zero or more flags, optional minimum field width and a conversion directive as follows:
+        /// <param name="specifier">the entire specifier, such as %Y for a 4-digit year</param>
+        /// <param name="flags">zero or more flags to change formatting, such as '0' (zero padding).</param>
+        /// <param name="width">optional minimum field width</param>
+        /// <param name="directive">The required data value, such as Y for a 4-digit year</param>
+        /// <param name="source">the source date-time object</param>
+        /// </summary>

```

Comment:
Misplaced end tag `/// </summary>`
The end tag for summary must be above the parameter elements

Reply:
Everything else looks good.